runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectRunner.java

```diff
@@ -130,9 +132,13 @@ static BundleFactory bundleFactoryFor(Set<Enforcement> enforcements, DirectGraph
   private final DirectOptions options;
   private final Set<Enforcement> enabledEnforcements;
   private Supplier<Clock> clockSupplier = new NanosOffsetClockSupplier();
+  private static final ObjectMapper MAPPER =
+      new ObjectMapper()
+          .registerModules(ObjectMapper.findModules(ReflectHelpers.findClassLoader()));
 
   /** Construct a {@link DirectRunner} from the provided options. */
   public static DirectRunner fromOptions(PipelineOptions options) {
+    options = MAPPER.convertValue(MAPPER.valueToTree(options), DirectOptions.class);
```

Comment:
`convertValue` / `valueToTree` may go through specialized paths which partially convert value types and allow for things like static references to stick around. Instead try using [ObjectMapper#readValue](https://fasterxml.github.io/jackson-databind/javadoc/2.2.0/com/fasterxml/jackson/databind/ObjectMapper.html#readValue(byte[],%20java.lang.Class)) and [ObjectMapper#writeValues](https://fasterxml.github.io/jackson-databind/javadoc/2.2.0/com/fasterxml/jackson/databind/ObjectMapper.html#writeValueAsBytes(java.lang.Object))

Can you also add a unit test?

Reply:
@lukecwik Thanks for the suggestions.
Added changes as suggested and also added unit test for the same. Please review my changes.